packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -139,60 +223,79 @@ export class VaFileInput {
 
       return (
         <Host>
-          {label && (
-            <label htmlFor="fileInputButton" class={labelClass} part="label">
-              {label}
-              {required && (
-                <span class="usa-label--required">
-                  {' '}
-                  {i18next.t('required')}
-                </span>
-              )}
-            </label>
-          )}
+          {label && this.renderLabelOrHeader(label, required, headerSize)}
           {hint && (
-            <span class="usa-hint" id="input-hint-message">
-              {hint}
-            </span>
+            <div class="usa-hint" id="input-hint-message">{hint}</div>
           )}
-          <slot></slot>
           <span id="input-error-message" role="alert">
-            {error && (
-              <Fragment>
-                <span class="usa-sr-only">{i18next.t('error')}</span>
-                <span class="usa-error-message">{error}</span>
-              </Fragment>
+             {error && (
+               <Fragment>
+                 <span class="usa-sr-only">{i18next.t('error')}</span>
+                 <span class="usa-error-message">{error}</span>
+               </Fragment>
+             )}
+           </span>
+          <div class="file-input-wrapper" onDrop={this.handleDrop}>
+            <input
+              id="fileInputField"
+              class="file-input"
+              style={{ visibility: this.uploadStatus === 'success' ? 'hidden' : 'unset' }}
+              type="file"
+              ref={el => this.fileInputRef = el as HTMLInputElement}
+              name={name}
+              accept={accept}
+              aria-describedby={ariaDescribedbyIds}
+              onChange={this.handleChange}
+            />
+            {uploadStatus === 'idle' && (
+              <div>
+                <div class="sr-only">No files selected.</div>
+                <div class={fileInputTargetClasses}>
+                  <div class="file-input-box"></div>
+                  <div class="file-input-instructions">
+                    <span class="file-input-drag-text">Drag files here or </span>
+                    <span class="file-input-choose-text">choose from folder</span>
+                  </div>
+                </div>
+              </div>
             )}
-          </span>
-          <input
-            id="fileInputField"
-            class={inputClass}
-            type="file"
-            name={name}
-            accept={accept}
-            aria-describedby={ariaDescribedbyIds}
-            onChange={this.handleChange}
-          />
+            {uploadStatus !== 'idle' && (
+              <div class="selected-files-wrapper">
+                <div class="selected-files-label">Selected files</div>
+                <va-card class="va-card">
+                  <div class="file-info-section">
+                    <va-icon icon="file_present" size={5} class="file-icon"></va-icon>
+                    <span class="file-label">{file.name}</span>
```

Comment:
@danbrady @LWWright7 - We need to decide on how to approach name truncation for files with long names. Would appreciate some design input here, and potentially an update to the figma specs to reflect name truncation. 

Currently long names extend past the card: 
![Screenshot from 2024-05-08 13-15-52](https://github.com/department-of-veterans-affairs/component-library/assets/15097156/3cc84bf6-728e-4a5e-8470-3249a7796f65)

Reply:
Ok, I think we can handle long file names by splitting the file name and the uploading label / file size to be on different lines. In other words, the file name will be on top line and then either the "Uploading..." label or the file weight on the next line. If the file name is long, it should wrap, and not truncate it, so it does not break outside the boundaries of the card. (This stays inline with what USWDS does.) Some designs below for reference:

### **Previous design**: 
(This didn't factor long file names)
<img width="453" alt="image" src="https://github.com/department-of-veterans-affairs/component-library/assets/381190/7eab6bbd-c97b-45ef-b353-bff47208915e">



---

### Updated design:
<img width="453" alt="image" src="https://github.com/department-of-veterans-affairs/component-library/assets/381190/157c9ead-b96d-4953-9385-d3851e02a8aa">

---

### Updated design with long name and in uploading state:
<img width="453" alt="image" src="https://github.com/department-of-veterans-affairs/component-library/assets/381190/d92001a1-d893-4f17-b3b4-7dd697dc3395">

---

### Updated design with long name attached:
<img width="493" alt="image" src="https://github.com/department-of-veterans-affairs/component-library/assets/381190/5b6c493e-a824-479a-88fb-64d1dfb1682a">

---

For Reference:
**USWDS with Long File Name**:
<img width="499" alt="Screenshot 2024-05-13 at 1 56 32 PM" src="https://github.com/department-of-veterans-affairs/component-library/assets/381190/5058628f-59a6-4b6f-9b76-e9e5d7c78ea2">


**NOTE**: The default body text has a line-height that's too much for this component. So instead I used [line-height-2](https://design.va.gov/foundation/utilities/line-height) (1.15) for the file name and size text, which renders 16.96px text to have 19.504 line-height. 

I've updated the design files with this. Hit my up with any questions!